datastore/api/snippets_test.py

```diff
@@ -11,291 +11,269 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from functools import wraps
 import time
 
 from gcloud import datastore
+import pytest
 import snippets
-from testing import CloudTest, mark_flaky
+from testing import mark_flaky
 
 
-def eventually_consistent(f):
-    @wraps(f)
-    def inner(self, *args, **kwargs):
-        # This is pretty hacky, but make datastore wait 1s after any
-        # put operation to in order to account for eventual consistency.
-        original_put_multi = self.client.put_multi
+class CleanupClient(datastore.Client):
+    def __init__(self, *args, **kwargs):
+        super(CleanupClient, self).__init__(*args, **kwargs)
+        self.entities_to_delete = []
+        self.keys_to_delete = []
```

Comment:
The problem is that not everything needs to be deleted. Some of the queries do thing with stats, namespaces, and other meta objects. The tests themselves add stuff to the lists, so there shouldn't be any issue for someone else running these tests.